testing/selenium_tests.py

```diff
@@ -0,0 +1,625 @@
+        dashboard_id = "ac1078e0-8a32-11ea-8939-89f508ff7909"
+        driver.get(f"https://ls1/app/dashboards#/view/{dashboard_id}")
+        expected_cond = EC.presence_of_element_located((By.CLASS_NAME, "react-grid-layout"))
+        WebDriverWait(driver, args.timeout).until(expected_cond)
+
+    def test_signals_overview(self):
+        """Is there any data?"""
+        panel = load_panel("Signals Overview")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_mitre_attack_technique(self):
+        """Is there any data?"""
+        panel = load_panel("MITRE ATT&CK Technique")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_signals_details(self):
+        """Is there any data?"""
+        panel = load_panel("Signals Details")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_full_event_logs(self):
+        """Is there any data?"""
+        panel = load_panel("Full Event Logs")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+class HealthCheckTests(unittest.TestCase):
+    """Test cases for the HealthCheck Dashboard"""
+    #2/6/2024, main branch on lme. The health check dashboard has an odd dashboard menu. This will likely need updating.
+
+    def setUp(self):
+        # The dashboard ID is hard-coded in the ndjson file
+        dashboard_id = "51fe1470-fa59-11e9-bf25-8f92ffa3e3ec"
+        driver.get(f"https://ls1/app/dashboards#/view/{dashboard_id}")
+        expected_cond = EC.presence_of_element_located((By.CLASS_NAME, "react-grid-layout"))
+        WebDriverWait(driver, args.timeout).until(expected_cond)
+
+    def test_total_hosts(self):
+        """Is there any data?"""
+        panel = load_panel("Alpha - Health Check - Total Hosts")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_users_seen(self):
+        """Is there any data?"""
+        panel = load_panel("Users seen")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_number_of_admins(self):
+        """Is there any data?"""
+        panel = load_panel("Alpha - Health Check - Number of Admins - Metric (converted)")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_events_by_machine(self):
+        """Is there any data?"""
+        panel = load_panel("Events by machine")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_unexpected_shutdowns(self):
+        """Is there any data?"""
+        panel = load_panel("Unexpected shutdowns")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+options = webdriver.ChromeOptions()
+# options.add_argument("--headless=new")
+options.add_experimental_option("detach", True) # Make browser stay open, in reality we'd
+# probably prefer headless

```

Comment:
Detached was nice for debugging, but as the comment says, for an actual testing workflow we'd probably prefer headless.

Recommend:
1. Testing to make sure headless works as expected
2. Adding a switch to the command line arguments that allows the user to specify if they want headless or detached

Reply:
Done